modules/core/src/main/java/org/apache/ignite/internal/util/IgniteObjectName.java

```diff
@@ -0,0 +1,140 @@
+
+import org.apache.ignite.lang.IgniteInternalException;
+
+/**
+ * Utility methods used for cluster's named objects: schemas, tables, columns, indexes, etc.
+ */
+public class IgniteObjectName {
+    /** No instance methods. */
+    private IgniteObjectName() {
+    }
+
+    /**
+     * Parse database's object name: unquote name or cast to upper case not-quoted name.
+     *
+     * @param str String to parse object name.
+     * @return Unquoted name or name is cast to upper case. "tbl0" -&gt; "TBL0", "\"Tbl0\"" -&gt; "Tbl0".
+     */
+    public static String parse(String str) {
+        if (str == null || str.isEmpty()) {
+            return str;
+        }
+
+        if (str.startsWith("\"") && str.endsWith("\"")) {
+            if (str.length() == 1) {
+                throw new IgniteInternalException("Invalid identifier: single quota");
+            }
+
+            return str.substring(1, str.length() - 1);
+        } else {
+            return str.toUpperCase();
+        }
+    }
+
+    /**
+     * Parse canonical table [schemaName].[tableName], unquote identifiers and normalize case, e.g. "public.tbl0" -&gt; "PUBLIC.TBL0",
+     * "PUBLIC.\"Tbl0\"" -&gt; "PUBLIC.Tbl0", "\"MySchema\".\"Tbl0\"" -&gt; "MySchema.Tbl0", etc.
+     *
+     * @param str String to parse canonical name.
+     * @return Unquote identifiers and normalize case.
+     */
+    public static String parseCanonicalName(String str) {
+        if (str == null || str.isEmpty()) {
+            throw new IgniteInternalException("Invalid identifier: empty string");
+        }
+
+        StringBuilder name = new StringBuilder();
+        boolean quoted = false;
+        int idBegin = 0;
+        int idEnd = 0;
+
+        for (int i = 0; i < str.length(); ++i) {
+            switch (str.charAt(i)) {
+                case '\"':
+                    if (quoted) {
+                        idEnd = i;
+                    } else {
+                        idBegin = i + 1;
+                        quoted = true;
+                    }
+
+                    break;
+
+                case '.':
+                    if (!quoted) {
+                        idEnd = i;
+                    }
+
+                    String id = str.substring(idBegin, idEnd);
+
+                    name.append(quoted ? id : id.toUpperCase()).append('.');
+
+                    quoted = false;
+                    idBegin = i + 1;
+
+                    break;
+
+                default:
+                    break;
+            }
+        }
+
+        if (!quoted) {
+            idEnd = str.length();
+        }
+
+        // append last name
+        String id = str.substring(idBegin, idEnd);
+
+        name.append(quoted ? id : id.toUpperCase());
+
+        return name.toString();
+    }
+
+    /**
+     * Unparse database's object name: quote name.
+     *
+     * @param str Object name.
+     * @return Quoted object name.
+     */
+    public static String unparseName(String str) {
```

Comment:
```suggestion
    public static String quote(String str) {
```

Reply:
If the `unparseName` is renamed to `quote` what name is applicable for `unparseCanonicalName`?
@AMashenkov